ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -491,34 +503,21 @@ func (ipfs *Connector) PinLs(ctx context.Context, typeFilter string) (map[string
 	return statusMap, nil
 }
 
-// PinLsCid performs a "pin ls <hash>" request. It first tries with
-// "type=recursive" and then, if not found, with "type=direct". It returns an
-// api.IPFSPinStatus for that hash.
-func (ipfs *Connector) PinLsCid(ctx context.Context, hash cid.Cid) (api.IPFSPinStatus, error) {
+// PinLsCid performs a "pin ls <hash>" request. It will use "type=recursive" or
+// "type=direct" (or other) depending on the given pin's MaxDepth setting.
+// It returns an api.IPFSPinStatus for that hash.
+func (ipfs *Connector) PinLsCid(ctx context.Context, pin *api.Pin) (api.IPFSPinStatus, error) {
```

Comment:
I changed the signature of PinLsCid so that we can get a full Pin object and we can check `direct` or `recursive` direclty, without having to trigger two requests for checking if a direct pin is pinned.

This affects the RPC api, but since this is a local-only endpoint, we don't need to bump the RPC protocol version.